Amplify/Categories/DataStore/Model/Internal/Schema/ModelSchema+Identifiers.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  ModelSchema+Identifiers.swift
+//  
+//
+//  Created by Sapir Muallem on 23/04/2023.
+//
+
+import Foundation
+
+extension ModelSchema {
+    func getModelIdentifiers(from modelObject: [String: JSONValue]) -> [LazyReferenceIdentifier] {
+        var identifiers = [LazyReferenceIdentifier]()
+
+        for identifierField in primaryKey.fields {
+            if case .string(let identifierValue) = modelObject[identifierField.name] {
+                identifiers.append(.init(name: identifierField.name, value: identifierValue))
+            }
```

Comment:
Since this relies on external JSON, ignoring all other types (`array`, `boolean`, `number`, etc.) silently may case unforeseen consequences. How about we `throw` with something like this?:

```
func lazyReferenceIdentifiers(from modelObject: [String: JSONValue]) throws -> [LazyReferenceIdentifier] {
    enum ExtractionError: Swift.Error {
        case unsupportedLazyReferenceIdentifier(name: String, value: JSONValue?)
    }

    var identifiers = [LazyReferenceIdentifier]()
    for identifierField in primaryKey.fields {
        let object = modelObject[identifierField.name]
        switch object {
        case .string(let identifierValue):
            identifiers.append(.init(name: identifierField.name, value: identifierValue))
        default:
            throw ExtractionError.unsupportedLazyReferenceIdentifier(
                name: identifierField.name,
                value: object
            )
        }
    }

    return identifiers
}
```

Reply:
great idea, thanks

Reply:
great idea, thanks